Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
+{
+	/**
+	 * @param int $parentId
+	 * @return void
+	 */
+	public function setParentId($parentId);
+	
+	/**
+	 * @return int
+	 */
+	public function getParentId();
+	
+	/**
+	 * @param int $questionId
+	 * @return void
+	 */
+	public function setId($questionId);
+	
+	/**
+	 * @return int $questionId
+	 */
+	public function getId();
+	
+	/**
+	 * @return string
+	 */
+	public function getTitle();
+	
+	/**
+	 * @return string
+	 */
+	public function getComment();
+	
+	/**
+	 * @return string
+	 */
+	public function getOwner();
+	
+	/**
+	 * @return string $questionType
+	 */
+	public function getQuestionType();
```

Comment:
This surely cannot be any string, right? Is there a list of these strings somewhere? Also: Do we need a second typisation of the questions? They already will have a class, that might be sufficient.

Reply:
question types are registered in the database. Plugin question types needs to be handled having this information stored nevertheless. At several places this methid is in use, so we need it, yes.

It might be removed in the future.